### PR TITLE
docker option to pick ACE source

### PR DESCRIPTION
**Pull request by adamsj-oci**
Use `ACE_CONFIG_OPTION` and `ACE_SOURCE_DIR` options to pick difference ACE source. 

`ACE_CONFIG_OPTION` can be used to select `--ace-github-latest` instead of the default `--doc-group`.

`ACE_SOURCE_DIR` copies the correct files based on the ACE source to `/usr/local/share/ace/MPC`. Defaults to `/opt/OpenDDS/ACE_wrappers/MPC`. Use `/opt/OpenDDS/ACE_TAO/ACE/MPC` when using `--ace-github-latest` in `ACE_CONFIG_OPTION`.

### CODE REVIEW

**Comment by iguessthislldo** [comment]
Why `ACE_wrappers/MPC`?

**Comment by iguessthislldo** [comment]
Actually forget my last comment, I now see you're copying MPC. I would suggest calling it `MPC_SOURCE_DIR` or maybe `MPC_ROOT`, which is the convention, instead.

**Comment by adamsj-oci** [comment]
> Why `ACE_wrappers/MPC`?

That much I didn't pick. That's the original path. I'm just adjusting it when the `--ace-github-latest` option is used in the docker build.

File: hooks/build [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 if [[ -z "${BASIS}" ]]; then
-    docker build -t $IMAGE_NAME .
+    docker build --build-arg ACE_CONFIG_OPTION=$ACE_CONFIG_OPTION --build-arg MPC_ROOT=$MPC_ROOT -t $IMAGE_NAME .
 else
-    docker build --build-arg BASIS=$BASIS -t $IMAGE_NAME .
+    docker build --build-arg BASIS=$BASIS --build-arg= ACE_CONFIG_OPTION=$ACE_CONFIG_OPTION --build-arg MPC_ROOT=$MPC_ROOT -t $IMAGE_NAME .
```
Comment (mitza-oci) [inline hooks/build]:
```suggestion
    docker build --build-arg BASIS=$BASIS --build-arg ACE_CONFIG_OPTION=$ACE_CONFIG_OPTION --build-arg MPC_ROOT=$MPC_ROOT -t $IMAGE_NAME .
```
just a guess...?

File: Dockerfile [hunk 2]
```diff
@@ -20,11 +21,13 @@ RUN cmake CMakeLists.txt && make && cp ./*.a /usr/lib
...
 
+ARG ACE_CONFIG_OPTION="--doc-group"
+ARG MPC_ROOT="/opt/OpenDDS/ACE_wrappers/MPC"
 RUN cd /opt/OpenDDS && \
-    ./configure --prefix=/usr/local --security --doc-group --std=c++11 && \
+    ./configure --prefix=/usr/local --security --std=c++11 ${ACE_CONFIG_OPTION} && \
     make && \
     make install && \
-    cp -a /opt/OpenDDS/ACE_wrappers/MPC /usr/local/share/ace/MPC
+    cp -a ${MPC_ROOT} /usr/local/share/ace/MPC
```
Comment (mitza-oci) [inline Dockerfile]:
```suggestion
    cp -a ${MPC_ROOT} /usr/local/share/MPC
```
This shouldn't be in `ace`, but we may want to keep compatibility with old docker image? @jrw972

File: hooks/build (hunk 1 above)
Comment (adamsj-oci) [reply hooks/build]: I just found this too. Fixed.

File: Dockerfile (hunk 2 above)
Comment (jrw972) [reply Dockerfile]: I don't think backward compatibility is an issue.  IMHO, it is highly unlikely that any use of the Docker image actually cares.

My only request is that after this PR is done, go through the "Compile the Messenger Example" of https://opendds.org/quickstart/GettingStartedDocker.html to make sure it still works.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -233,7 +233,10 @@ jobs:
         ConfigOpts: --security --features=versioned_namespace=1
         PackageDeps: libxerces-c-dev libssl-dev cmake
       SecurityWithoutFeatures:
-        ConfigOpts: --no-inline --no-debug --no-built-in-topics --no-content-subscription --no-ownership-profile --no-object-model-profile --no-persistence-profile --security
+        ConfigOpts: >
+          --security --no-inline --no-debug
+          --no-built-in-topics --no-content-subscription --no-ownership-profile
```
Comment (neeleym) [inline azure-pipelines.yml]: I don't think that we should be using --no-built-in-topics as those are needed for dynamic discovery aspects. Has there been prior discussion to drop them out for some reason?

Comment (mitza-oci) [reply azure-pipelines.yml]: That's not part of Jeremy's changes. Jeremy - this branch now contains rewritten commits, please avoid this.  Merge again and force-push to reset your branch.

Comment (neeleym) [reply azure-pipelines.yml]: Understood ... ignore my question.

File: Dockerfile (hunk 2 above)
Comment (mitza-oci) [reply Dockerfile]: I don't know the answer to these, but I'm in favor of `/usr/local/share/MPC`.

**Comment by mitza-oci** [comment]
I don't know what's going on with this PR, but your source branch looks OK.  Maybe we just need to close it and open a new PR for the same source.

File: azure-pipelines.yml (hunk 3 above)
Comment (adamsj-oci) [reply azure-pipelines.yml]: Gotcha. I didn't use a separate branch for this thinking it was a quick change.

**Pull request closed without merging**